core/src/main/java/org/radargun/stats/Statistics.java

```diff
@@ -173,9 +175,59 @@ default Statistics with(Statistics otherStats) {
     */
    <T> T getRepresentation(String operation, Class<T> clazz, Object... args);
 
+   StreamOfRequest getStreamOfRequest();
+
    class Converter extends ReflexiveConverters.ObjectConverter {
       public Converter() {
          super(Statistics.class);
       }
    }
+
+   class UniformRateLimiter {
+      long start = Long.MIN_VALUE;
+      final long intervalNs;
+      final AtomicLong opIndex = new AtomicLong();
+
+      UniformRateLimiter(int opsPerSec) {
+         intervalNs = 1000000000 / opsPerSec;
+      }
+
+      /* @return intended start time in ns for the operation */
+      long acquire() {
+         long currOpIndex = opIndex.getAndAdd(1);
+         return start + currOpIndex * intervalNs;
```

Comment:
As `intervalNs` is be rounded down the time the operation was supposed to start will slowly drift down. You need to use `start + currOpIndex * 1000_000_000 / opsPerSec` to multiply first.